src/Input/CardDisplayInput.jsx

```diff
@@ -0,0 +1,130 @@
+import React, { useMemo } from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import propTypes from '@styled-system/prop-types'
+import css from '@styled-system/css'
+import { variant } from 'styled-system'
+import { Icon } from '../Icon'
+import { Typography, typography } from '../Typography'
+import { Flex, Touchable } from '@ivoryio/kogaio'
+
+import { default as variants } from './CardDisplayInput.variants'
+import { SubLabel } from './SubLabel'
+import sizes from '../assets/sizes'
+
+export const CardDisplayInput = ({
+  disabled,
+  id,
+  label,
+  name,
+  noBottomSpace,
+  onEdit,
+  value,
+  variant,
+  subLabel,
+  ...rest
+}) => {
+  const inputVariant = useMemo(() => {
+    if (disabled) return 'disabled'
+    return variant
+  }, [disabled, variant])
+
+  const handleEdit = () => {
+    if (!disabled) {
+      onEdit()
+    }
+  }
+
+  return (
+    <Flex flexDirection='column' hasLabel={label} width={1} {...rest}>
+      {label ? (
+        <Typography
+          as='label'
+          htmlFor={id}
+          variant='inputLabel'
+          width='fit-content'>
+          {label}
+        </Typography>
+      ) : null}
+      <Flex alignItems='center' position='relative' width='100%'>
+        <StyledInput
+          ariaDisabled={disabled}
+          hasLabel={label}
+          hasIcRight={true}
+          id={id}
+          name={name}
+          readOnly={true}
+          value={`●●●● ●●●● ●●●● ${value || '0000'}`}
+          variant={inputVariant}
+          {...rest}
+        />
+        <Flex pointerEvents='auto' position='absolute' right={2}>
+          <StyledTouchable onClick={handleEdit}>
+            <Icon mr={5} size='small' name='edit' color='inputIcon' />
+          </StyledTouchable>

```

Comment:
If the CardInput is disabled, this icon should reflect th state by color. Also, no pointer events on the disabled state